src/BSON/PackedArray.c

```diff
@@ -401,6 +421,48 @@ static HashTable* php_phongo_packedarray_get_properties(phongo_compat_object_han
 	return php_phongo_packedarray_get_properties_hash(object, false, 1);
 }
 
+zval* php_phongo_packedarray_read_dimension(phongo_compat_object_handler_type* object, zval* offset, int type, zval* rv)
+{
+	php_phongo_packedarray_t* intern;
+
+	intern = Z_OBJ_PACKEDARRAY(PHONGO_COMPAT_GET_OBJ(object));
+
+	if (Z_TYPE_P(offset) != IS_LONG) {
+		phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not find index of type \"%s\" in BSON data", PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(offset));
+		return &EG(uninitialized_zval);
+	}
+
+	if (!php_phongo_packedarray_get(intern, Z_LVAL_P(offset), rv)) {
+		// Exception already thrown
+		return &EG(uninitialized_zval);
+	}
+
+	return rv;
+}
+
+void php_phongo_packedarray_write_dimension(phongo_compat_object_handler_type* object, zval* offset, zval* value)
+{
+	phongo_throw_exception(PHONGO_ERROR_LOGIC, "Cannot write to %s offset", ZSTR_VAL(php_phongo_packedarray_ce->name));
+}
+
+int php_phongo_packedarray_has_dimension(phongo_compat_object_handler_type* object, zval* member, int check_empty)
+{
+	php_phongo_packedarray_t* intern;
+
+	intern = Z_OBJ_PACKEDARRAY(PHONGO_COMPAT_GET_OBJ(object));
+
+	if (Z_TYPE_P(member) != IS_LONG) {
+		return false;
```

Comment:
I assume it's intentional to not throw for an invalid type, as you do in `get`. Would doing so mess up `isset` and the like?

Reply:
Not really, I just figured that we can be more lenient in `isset` checks since I'd assume people use those methods to guard access to the fields in the BSON structure, in which case returning `false` instead of throwing makes more sense.